experimental/packages/opentelemetry-exporter-prometheus/src/PrometheusSerializer.ts

```diff
@@ -168,112 +176,131 @@ export class PrometheusSerializer {
-      }
-      case AggregatorKind.HISTOGRAM: {
-        const { value, timestamp: hrtime } = record.aggregator.toPoint();
-        const timestamp = hrTimeToMilliseconds(hrtime);
-        /** Histogram["bucket"] is not typed with `number` */
-        for (const key of ['count', 'sum'] as ('count' | 'sum')[]) {
-          results += stringify(
-            name + '_' + key,
-            record.attributes,
-            value[key],
-            this._appendTimestamp ? timestamp : undefined,
-            undefined
-          );
-        }
+  serializeHistogramDataPoint(name: string, type: InstrumentType, dataPoint: DataPoint<Histogram>): string {
+    let results = '';
 
-        let cumulativeSum = 0;
-        const countEntries = value.buckets.counts.entries();
-        let infiniteBoundaryDefined = false;
-        for (const [idx, val] of countEntries) {
-          cumulativeSum += val;
-          const upperBound = value.buckets.boundaries[idx];
-          /** HistogramAggregator is producing different boundary output -
-           * in one case not including inifinity values, in other -
-           * full, e.g. [0, 100] and [0, 100, Infinity]
-           * we should consider that in export, if Infinity is defined, use it
-           * as boundary
-           */
-          if (upperBound === undefined && infiniteBoundaryDefined) {
-            break;
-          }
-          if (upperBound === Infinity) {
-            infiniteBoundaryDefined = true;
-          }
-          results += stringify(
-            name + '_bucket',
-            record.attributes,
-            cumulativeSum,
-            this._appendTimestamp ? timestamp : undefined,
-            {
-              le:
-                upperBound === undefined || upperBound === Infinity
-                  ? '+Inf'
-                  : String(upperBound),
-            }
-          );
-        }
+    name = enforcePrometheusNamingConvention(name, type);
+    const { value, attributes } = dataPoint;
+    const timestamp = hrTimeToMilliseconds(dataPoint.endTime);
+    /** Histogram["bucket"] is not typed with `number` */
+    for (const key of ['count', 'sum'] as ('count' | 'sum')[]) {
+      results += stringify(
+        name + '_' + key,
+        attributes,
+        value[key],
+        this._appendTimestamp ? timestamp : undefined,
+        undefined
+      );
+    }
+
+    let cumulativeSum = 0;
+    const countEntries = value.buckets.counts.entries();
+    let infiniteBoundaryDefined = false;
+    for (const [idx, val] of countEntries) {
+      cumulativeSum += val;
+      const upperBound = value.buckets.boundaries[idx];
+      /** HistogramAggregator is producing different boundary output -
+       * in one case not including infinity values, in other -
+       * full, e.g. [0, 100] and [0, 100, Infinity]
+       * we should consider that in export, if Infinity is defined, use it
+       * as boundary
+       */
+      if (upperBound === undefined && infiniteBoundaryDefined) {
```

Comment:
Shouldn't HistogramAggregator always be implicitly including infinity as the last upper bound (overflow bucket)?

For prometheus specifically, the `+Inf` boundary [must always be present](https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md#:~:text=A%20histogram%20must%20have%20a%20bucket%20with%20%7Ble%3D%22%2BInf%22%7D.%20Its%20value%20must%20be%20identical%20to%20the%20value%20of%20x_count.)

Reply:
Ah, looks like this is handling both cases so user can explicitly or implicitly pass the infinity bucket when creating the Histogram aggregation in a view?

Reply:
Yeah, right. The HistogramAggregator itself doesn't implicitly add an `Infinity` upper bound -- instead, there is an additional value slot to represent counts of values greater than the explicitly defined upper bound:

```
boundaries: [        0,   10,    100,     1000]
values:     [ -Inf - 0, 1-10, 10-100, 100-1000, 1000- +Inf]   
```

And for explicit `+Infinity` upper bound:

```
boundaries: [        0,   10,    100,     1000,       +Inf]
values:     [ -Inf - 0, 1-10, 10-100, 100-1000, 1000- +Inf]   
```

Reply:
Gotcha. Outside the scope of this PR, but it would be nice if the aggregation only ever returns the "OTLP way" with implicit under/overflow bounds (so exporters don't have to handle both)

Reply:
Yeah, SGTM. I'll work on this in another PR.

Reply:
Submitted https://github.com/open-telemetry/opentelemetry-js/pull/2859 for this.